### ISSUE TITLE
Code for user-initialized setting

### ISSUE THREAD

**Issue by Flowerfan** [TAO-Dataset/tao#23]
Hi Achal,

The paper reports the results for user-initialized methods. Could you please release related code?

**Comment by achalddave** [comment]
Yes, of course! I've been meaning to do this for a while and kept putting it off. I've quickly pushed my code for running single-object trackers based on the [pysot](https://github.com/STVIR/pysot) repo, with documentation for the steps here: https://github.com/TAO-Dataset/tao/blob/master/docs/trackers.md#single-object-trackers

To use your own tracker, you can do the following:
1. Implement your tracker by inheriting from the base [Tracker](https://github.com/TAO-Dataset/tao/blob/master/tao/trackers/sot/base.py) class. You can look at how the PysotTracker is implemented [here](https://github.com/TAO-Dataset/tao/blob/master/tao/trackers/sot/pysot.py).
2. Update `init_tracker` [here](https://github.com/TAO-Dataset/tao/blob/master/scripts/trackers/single_obj/pysot_trackers.py#L26-L39) and add an option for your tracker [here](https://github.com/TAO-Dataset/tao/blob/master/scripts/trackers/single_obj/pysot_trackers.py#L133) to add support for your tracker.

I did this fairly hastily so please let me know if you run into any issues!

**Comment by Flowerfan** [comment]
Hi, @achalddave 

Thank you for your great job!

I have tested the SiamRPN++ model using the first frame as the init frame following the instructions. However, I only get 25.34% mAP on the TAO validation set (988 videos), which is much lower than 29.7% reported in Table 5.  The model I have downloaded is siamrpn_r50_l234_dwxcorr from [pysot zoo](https://github.com/STVIR/pysot/blob/052b9678a7ed336752f74dc6af31cc00eb004551/MODEL_ZOO.md). 
Are there any hyperparameters I should tune to reproduce the result?

**Comment by achalddave** [comment]
That is strange. We did not tune hyperparameters, to my recollection. I'll look into this soon and get back to you. If possible, could you send me your results file, either here or over email?

**Comment by Flowerfan** [comment]
> That is strange. We did not tune hyperparameters, to my recollection. I'll look into this soon and get back to you. If possible, could you send me your results file, either here or over email?

I have attached the result file to [google drive here](https://drive.google.com/file/d/1vbKlGfEh4XfuERrZqkL6KBVGDDXpIfMD/view?usp=sharing). Please check it when you are available. Thank you.

**Comment by achalddave** [comment]
Ah, my bad, there is one key hyperparameter that requires tuning: a confidence threshold. Single object trackers report boxes for all frames, even if they are underconfident in some frames. Thus, you need to specify a confidence threshold, and remove predictions for frames with low confidences.

To do this, we tuned the score threshold on the training set, as described in Table 16 of the supplementary:
<img width="728" alt="Screen Shot 2021-01-31 at 12 09 58 PM" src="https://user-images.githubusercontent.com/1168881/106391835-3ee38900-63bd-11eb-80d1-05fe9bd2a665.png">

Your result of 28.4 on training set is pretty close to my result with a score threshold of 0 (28.6AP). Can you try evaluating on the validation set using a score threshold of 0.7, which is the threshold we found was optimal on the train set as shown in the above table? Once your results match, I will update the docs as well.

**Comment by Flowerfan** [comment]
I got the same performance with the threshold. It is very much appreciated.
